hack/docs/Adding_Detectors_external.md

```diff
@@ -62,15 +62,29 @@ If you think that something should be included outside of these guidelines, plea
    1. Create a [test secrets file, and export the variable](#using-a-test-secret-file)
    2. Update the pattern regex and keywords. Try iterating with [regex101.com](http://regex101.com/).
    3. Update the verifier code to use a non-destructive API call that can determine whether the secret is valid or not.
-   4. Update the tests with these test cases at minimum:
+      * Make sure you understand [verification indeterminacy](#verification-indeterminacy).
+   5. Update the tests with these test cases at minimum:
       1. Found and verified (using a credential loaded from GCP Secrets)
-      2. Found and unverified
-      3. Not found
-      4. Any false positive cases that you come across
+      2. Found and unverified (determinately, i.e. the secret is invalid)
+      3. Found and unverified (indeterminately due to timeout)
+      4. Found and unverified (indeterminately due to an unexpected API response)
+      5. Not found
+      6. Any false positive cases that you come across
    5. Create a pull request for review.
 
 ## Addendum
 
+### Verification indeterminacy
+
+There are two types of reasons that secret verification can fail:
+* The candidate secret is not actually a valid secret.
+* Something went wrong in the process unrelated to the candidate secret, such as a transient network error or an unexpected API response.
+
+In Trufflehog parlance, the first type of verification response is called _determinate_ and the second type is called _indeterminate_. Verification code should distinguish between the two by returning an error object in the result struct **only** for indeterminate failures. In general, a verifier should return an error (indicating an indeterminate failure) in all cases that haven't been explicitly identified as determinate failure states. For example, if a verifier makes an HTTP request to a hypothetical authentication endpoint, the response codes could determine what to return:
+* Any `2xx` response would indicate that verification succeeded.
+* A `403` response would indicate that verification failed **determinately** and no error object should be returned.
+* Any other response would indicate that verification failed **indeterminately** and an error object should be returned.
```

Comment:
I'm not sure we can say this is true for all apis we test against. e.g. Docker can still be a valid credential and return a 401 - https://github.com/trufflesecurity/trufflehog/blob/8ec5e4916c911144aab39e0602dfa930d3aa2d09/pkg/detectors/dockerhub/dockerhub.go#L79-L82

So maybe adjust the language to be less certain about the indeterminate verification failure in the case of non 2XX or 403 responses.

Reply:
I was trying to pick language that made it clear that was a hypothetical case, but I guess I didn't do a good job :(

Reply:
Maybe a separate paragraph for the start of the hypothetical case? And potentially changing the `would` to `would likely`?